File: packages/rook_sdk_apple_health/ios/Classes/Mapper/SyncTypeMappers.swift
//
//  SyncTypeMappers.swift
//  rook_sdk_apple_health
//
//  Created by Daniel Nolasco on 04/03/25.
//

import Foundation
import RookSDK

extension SummarySyncTypeProto {
    func toSyncType() -> String {
        switch self {
        case .sleep:
            return "sleep"
        case .physical:
            return "physical"
        case .body:
            return "body"
        default:
            fatalError("Unsupported summary type")
        }
    }
}

extension EventSyncTypeProto {
    func toSyncType() -> EventTypeToUpload {
        switch self {
        case EventSyncTypeProto.activity:
            return EventTypeToUpload.activityEvent
        case EventSyncTypeProto.bloodGlucose:
            return EventTypeToUpload.bloodGlucose
        case EventSyncTypeProto.bloodPressure:
            return EventTypeToUpload.bloodPressure
        case EventSyncTypeProto.bodyMetrics:
            return EventTypeToUpload.bodyMetrics
        case EventSyncTypeProto.heartRate:
            return EventTypeToUpload.heartRate
        case EventSyncTypeProto.oxygenation:
            return EventTypeToUpload.oxygenation
        case EventSyncTypeProto.temperature:
            return EventTypeToUpload.temperature
        default:
            fatalError("Unsupported event type")
        }
    }
}

File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/SyncTypeMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.model.SyncType
import com.rookmotion.rook_sdk_health_connect.exception.UnknownSyncTypeException
import com.rookmotion.rook_sdk_health_connect.proto.EventSyncTypeProto
import com.rookmotion.rook_sdk_health_connect.proto.SummarySyncTypeProto

fun SummarySyncTypeProto.toSyncType(): SyncType.Summary {
    return when (this) {
        SummarySyncTypeProto.SLEEP -> SyncType.Summary.SLEEP
        SummarySyncTypeProto.PHYSICAL -> SyncType.Summary.PHYSICAL
        SummarySyncTypeProto.BODY -> SyncType.Summary.BODY
        else -> throw UnknownSyncTypeException()
    }
}

fun EventSyncTypeProto.toSyncType(): SyncType.Event {
    return when (this) {
        EventSyncTypeProto.ACTIVITY -> SyncType.Event.ACTIVITY
        EventSyncTypeProto.BLOOD_GLUCOSE -> SyncType.Event.BLOOD_GLUCOSE
        EventSyncTypeProto.BLOOD_PRESSURE -> SyncType.Event.BLOOD_PRESSURE
        EventSyncTypeProto.BODY_METRICS -> SyncType.Event.BODY_METRICS
        EventSyncTypeProto.HEART_RATE -> SyncType.Event.HEART_RATE
        EventSyncTypeProto.HYDRATION -> SyncType.Event.HYDRATION
        EventSyncTypeProto.NUTRITION -> SyncType.Event.NUTRITION
        EventSyncTypeProto.OXYGENATION -> SyncType.Event.OXYGENATION
        EventSyncTypeProto.TEMPERATURE -> SyncType.Event.TEMPERATURE
        EventSyncTypeProto.STEPS -> SyncType.Event.STEPS
        EventSyncTypeProto.CALORIES -> SyncType.Event.CALORIES
        else -> throw UnknownSyncTypeException()
    }
}
